pkg/controllers/vdb/obj_reconciler.go

```diff
@@ -233,7 +233,7 @@ func (o *ObjReconciler) checkForDeletedSubcluster(ctx context.Context) (ctrl.Res
 	finder := iter.MakeSubclusterFinder(o.VRec.Client, o.Vdb)
 
 	// Find any statefulsets that need to be deleted
-	stss, err := finder.FindStatefulSets(ctx, iter.FindNotInVdb)
+	stss, err := finder.FindStatefulSets(ctx, iter.FindNotInVdb, vapi.MainCluster)
```

Comment:
Rather than hard coding the main cluster, could we use the sandbox found in the podfacts?